test/nightly/deploy.test.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { WebSiteManagementModels } from 'azure-arm-website';
+import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses, getRandomHexString, requestUtils } from '../../extension.bundle';
+import { longRunningTestsEnabled, testUserInput } from '../global.test';
+import { resourceGroupsToDelete, webSiteClient } from './global.resource.test';
+
+suite('Create Web App and deploy', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(20 * 6 * 1000);
+
+    suiteSetup(async function (this: IHookCallbackContext): Promise<void> {
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+    });
+
+    test('Node LTS', async () => {
+        const testFolderPath: string = await getWorkspacePath('nodejs-docs-hello-world');
+        await testCreateWebAppAndDeploy(['Linux', 'Node LTS'], testFolderPath);
+    });
+
+    async function testCreateWebAppAndDeploy(options: string[], workspacePath: string): Promise<void> {
+        const resourceName: string = getRandomHexString();
+        const resourceGroupName: string = getRandomHexString();
+        resourceGroupsToDelete.push(resourceGroupName);
+        const testInputs: (string | RegExp)[] = [resourceName, '$(plus) Create new resource group', resourceGroupName, ...options, '$(plus) Create new App Service plan', resourceName, 'B1', '$(plus) Create new Application Insights resource', resourceName, 'West US'];
+        await testUserInput.runWithInputs(testInputs, async () => {
+            await vscode.commands.executeCommand('appService.CreateWebAppAdvanced');
+        });
+        const createdApp: WebSiteManagementModels.Site = await webSiteClient.webApps.get(resourceGroupName, resourceName);
+        assert.ok(createdApp);
+
+        // Verify that the deployment is successful
+        await testUserInput.runWithInputs([workspacePath, resourceName, 'Deploy', DialogResponses.yes.title], async () => {
+            await vscode.commands.executeCommand('appService.Deploy');
+        });
+        const request: requestUtils.Request = await requestUtils.getDefaultRequest(`https://${resourceName}.azurewebsites.net`);
```

Comment:
Rather than hardcoding the url, could you retrieve the node and use `this.root.client.defaultHostUrl` to get the website?  This will help if we start testing with other endpoints like Mooncake.

Alternatively, we can update createWebApp and createWebAppAdvanced to return the `WebAppTreeItem` after it is called or return the WebAppTreeItem's ID and use that to retrieve the node like Functions https://github.com/microsoft/vscode-azurefunctions/blob/f641c1166f74ad11be2fc67b3e689d26ae8e40b0/src/commands/createFunctionApp/createFunctionApp.ts#L12-L29